browser-sign-in-kotlin/app/src/main/java/com/okta/browser/MainActivity.kt

```diff
@@ -52,6 +61,7 @@ class MainActivity : AppCompatActivity(), SharedPreferences.OnSharedPreferenceCh
 
     private var webAuthClient: WebAuthClient? = null
     private var authClient: AuthClient? = null
+    private var authenticationClient: AuthenticationClient? = null
```

Comment:
why are we always ignoring the null-safety aspects of kotlin? That's a main draw of the language, we should take advantage unless technically necessary.

> Normally, properties declared as having a non-null type must be initialized in the constructor. However, fairly often this is not convenient. For example, properties can be initialized through dependency injection, or in the setup method of a unit test. In this case, you cannot supply a non-null initializer in the constructor, but you still want to avoid null checks when referencing the property inside the body of a class.
> To handle this case, you can mark the property with the lateinit modifier

Reply:
I combined WebAuthClient and authClient into the same application. Since most of the code is the same except the sign in part.

WebAuthClient - sign in with browser.
AuthClient - custom sign in.

So depending on the settings option one of the clients will be null.